047-TrafficControlWithDapr/Student/Challenge-00.md

```diff
@@ -0,0 +1,323 @@
+
+    ```shell
+    az login
+    ```
+
+1.  Make sure you have selected the Azure subscription in which you want to work. Replace the 'x's with your subscription GUID or subscription name. The subscription GUID can be found in the Azure Resource Group blade from the Azure Portal.
+
+    ```shell
+    az account set --subscription "xxxx-xxxx-xxxx-xxxx"
+    ```
+
+1.  Generate an [SSH key pair](https://docs.microsoft.com/azure/virtual-machines/linux/ssh-from-windows#overview-of-ssh-and-keys) if you don't already have one.
+
+    ```shell
+    ssh-keygen -t rsa -b 2048
+    ```
+    - If prompted for a file name, leave the entry blank, and press enter.
+    - If prompted for a passphrase, leave the entry blank, and press enter.
+   
+    Once complete, you'll find two SSH key files in the following directory: `%USERPROFILE%\.ssh`. Right-click on the `id_rsa` file and open with Notepad. Copy the entire contents of the file which is the public key. You'll need it to configure the parameter file in an upcoming step.
+
+1.  In the accompanying source code, modify the `Resources/Infrastructure/bicep/main.parameters.json` file so it contains the proper data for the deployment:
+
+    ```json
+    {
+      "appName": {
+         "value": "dapr"
+      },
+      "region": {
+         "value": "southcentralus"
+      },
+      "environment": {
+         "value": "youruniqueid123"
+      },
+      "adminUserName": {
+         "value": "adminbruce"
+      },
+      "publicSSHKey": {
+         "value": "ssh-rsa AAAAB...wnBTn bruce.wayne@<your computer name>"        
+      }
+    }
+    ```
+
+1.  Create a new resource group for your WhatTheHack using the `Resources/Infrastructure/bicep/rg.bicep` script file. 
+
+    **When invoking the command, replace the `location` argument with the Azure region you want to use.**
+
+    ```shell
+    cd ./Resources/Infrastructure/bicep/
+    az deployment sub create --location "<resource-group-location>" --template-file rg.bicep --parameters ./main.parameters.json --query "properties.outputs" --output yamlc
+    ```
+
+1.  Upon completion, the command will display the name of the newly-created resource group:
+
+    ```yaml
+    resourceGroupName:
+    type: String
+    value: rg-dapr-youruniqueid123
+    ```
+
+    Copy the resource group as you'll use in the next command.
+
+1.  You'll now create the required Azure resources inside the new resource group with the following Azure CLI command.
+
+    ```shell
+    az deployment group create --resource-group "rg-dapr-youruniqueid123" --template-file main.bicep --parameters ./main.parameters.json --query "properties.outputs" --output yaml
+    ```
+
+    *Creating the resources can take some time. You're encouraged to jump to the **challenge 01** while the command executes.*
```

Comment:
Please change this to suggest students should review the Traffic app architecture instead of jumping to C1.